pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -298,8 +297,9 @@ func (vs *VSphere) cleanUpDummyVMs(dummyVMPrefix string) {
 			continue
 		}
 		// A write lock is acquired to make sure the cleanUp routine doesn't delete any VM's created by ongoing PVC requests.
-		defer cleanUpDummyVMLock.Lock()
+		cleanUpDummyVMLock.Lock()
 		err = diskmanagers.CleanUpDummyVMs(ctx, vmFolder, dc)
+		cleanUpDummyVMLock.Unlock()
```

Comment:
If CleanUpDummyVMs panics, this will never unlock. Is there a reason not to `defer cleanUpDummyVMLock.Unlock()` immediately after locking?

Reply:
The cleanUpDummyVMLock.Lock() is in a loop and not return, so we can not use defer.

Reply:
A typical pattern is to make an inner function and call it immediately, so that defer can be used